src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -119,24 +119,61 @@ def __init__(self, ws: WorkspaceClient, install_state: InstallState, verify_time
         # this dunder variable is hiding this method from tracebacks, making it cleaner
         # for the user to see the actual error without too much noise.
         __tracebackhide__ = True  # pylint: disable=unused-variable
         job_id = int(self._install_state.jobs[step])
         logger.debug(f"starting {step} job: {self._ws.config.host}#job/{job_id}")
         job_initial_run = self._ws.jobs.run_now(job_id)
-        if job_initial_run.run_id and not skip_job_wait:
-            try:
-                self._ws.jobs.wait_get_run_job_terminated_or_skipped(run_id=job_initial_run.run_id)
-            except OperationFailed as err:
-                logger.info('---------- REMOTE LOGS --------------')
-                self._relay_logs(step, job_initial_run.run_id)
-                logger.info('---------- END REMOTE LOGS ----------')
-                job_run = self._ws.jobs.get_run(job_initial_run.run_id)
-                raise self._infer_error_from_job_run(job_run) from err
+        run_id = job_initial_run.run_id
+        if not run_id:
+            raise NotFound(f"job run not found for {step}")
+        run_url = f"{self._ws.config.host}#job/{job_id}/runs/{run_id}"
+        logger.info(f"Started {step} job: {run_url}")
+        if skip_job_wait:
+            return run_id
+        try:
+            logger.debug(f"Waiting for completion of {step} job: {run_url}")
+            job_run = self._ws.jobs.wait_get_run_job_terminated_or_skipped(run_id=run_id, timeout=max_wait)
+            self._log_completed_job(step, run_id, job_run)
+            return run_id
+        except TimeoutError:
+            logger.warning(f"Timeout while waiting for {step} job to complete: {run_url}")
+            logger.info('---------- REMOTE LOGS --------------')
+            self._relay_logs(step, run_id)
+            logger.info('------ END REMOTE LOGS (SO FAR) -----')
+            raise
+        except OperationFailed as err:
+            logger.info('---------- REMOTE LOGS --------------')
+            self._relay_logs(step, run_id)
+            logger.info('---------- END REMOTE LOGS ----------')
+            job_run = self._ws.jobs.get_run(run_id)
+            raise self._infer_error_from_job_run(job_run) from err
+
+    @staticmethod
+    def _log_completed_job(step: str, run_id: int, job_run: Run) -> None:
+        if not logger.isEnabledFor(logging.INFO):
             return
-        raise NotFound(f"job run not found for {step}")
+        if job_run.state:
+            result_state = job_run.state.result_state or "N/A"
+            state_message = job_run.state.state_message
+            state_description = f"{result_state} ({state_message})" if state_message else f"{result_state}"
+            logger.info(f"Completed {step} job run {run_id} with state: {state_description}")
+        else:
+            logger.info(f"Completed {step} job run {run_id}, but end state unknown.")
```

Comment:
```suggestion
            logger.info(f"Completed {step} job run {run_id} with unknown state.")
```

Or rewrite the above to be `state_description = "UNKNOWN"` when `not job_run.state`

Reply:
I've adjusted the logging here: it's now a warning because this is an abnormal situation. The wording has also been tweaked. Does it now look better?

I distinctly want different wording for this compared to the normal situation: in general each logging statement should be distinct so that a log-entry can unambiguously traced to the line on which it was generated; this makes troubleshooting much easier.